### PR TITLE
bugfix: check alsadriver exist before cleanup

### LINKED ISSUE #13739: [MU4 Issue] Crash on changing buffer size in I/O Preferences
**Describe the bug**
MU4 crashes when you change buffer size in I/O Preferences

**To Reproduce**
1. Go to Preferences
2. Click on I/O
3. Change Buffer size
4. MU4 crashes

**Expected behavior**
MU4 should not crash

**Screenshots**

https://user-images.githubusercontent.com/88136054/195647134-a70f7659-ee8e-4f6e-b29c-77cff5f91eab.mp4

**Platform information**
 - OS: Ubuntu 22.04 LTS

### PR DESCRIPTION
**Pull request by lyrra**
resolves #13739

- [x] I signed [CLA](https://musescore.org/en/cla)

### CODE REVIEW

File: src/framework/audio/internal/platform/lin/linuxaudiodriver.cpp [hunk 1]
```diff
@@ -92,8 +92,11 @@ static void alsaCleanup()
     if (s_alsaData->threadHandle) {
         pthread_join(s_alsaData->threadHandle, nullptr);
     }
-    snd_pcm_drain(s_alsaData->alsaDeviceHandle);
-    snd_pcm_close(s_alsaData->alsaDeviceHandle);
+    qDebug("alsa handle: %p", s_alsaData->alsaDeviceHandle);
```
Comment (cbjeukendrup) [inline src/framework/audio/internal/platform/lin/linuxaudiodriver.cpp:95]: I think using qDebug here is not preferred. Better use `LOGD() << "alsa handle: " << s_alsaData->alsaDeviceHandle`, or remove the logging if it is not necessary.

Comment (lyrra) [reply src/framework/audio/internal/platform/lin/linuxaudiodriver.cpp]: Thanks! Forgot to remove debugging.

**Comment by HemantAntony** [comment]
@lyrra The issue still occurs in this build :(

**Comment by lyrra** [comment]
> @lyrra The crash still occurs in this build :(

Ouch! Would you mind running in gdb to get a stacktrace?

**Comment by HemantAntony** [comment]
I tried doing that, but there is no stack trace. Otherwise I would have provided it. I'll try again tomorrow :)

**Comment by lyrra** [comment]
I'm running this script to use either gdb or valgrind, running it at git-root after build is done.
sh run.sh g   -- would start gdb
sh run.sh v   -- would start valgrind

I tried it, changed the buffer-size, and got signal SIG32 in gdb (supposedly from the thread-cancels), just continue in gdb a few times would bring back the GUI again and no crash. 

```
#!/bin/sh

set -e

cd builds/Linux-*Make-Debug/install

export QT_QUICK_BACKEND="software"

G=""
V=""

if [ "$1" = "g" ]; then
  shift
  G="gdb --args"
elif [ "$1" = "v" ]; then
  shift
  V="valgrind --error-limit=no"
fi

$V $G ./bin/mscore $*

```

**Comment by RomanPudashkin** [comment]
Closing because the original issue is resolved

**Pull request closed without merging**